DasharoPayloadPkg/BlSupportPei/BlSupportPei.c

```diff
@@ -591,32 +582,22 @@ BlPeiEntryPoint (
   //
   // Parse memory info
   //
-  PldMemInfo.UsableLowMemTop = 0;
-  PldMemInfo.SystemLowMemTop = 0xFFFFFFFFUL;
-  Status = ParseMemoryInfo (MemInfoCallback, &PldMemInfo);
+  Status = ParseMemoryInfo (MemInfoCallback, &UsableLowMemTop);
   if (EFI_ERROR(Status)) {
     return Status;
   }
 
   //
   // Install memory
   //
-  LowMemorySize = PldMemInfo.UsableLowMemTop;
-  ASSERT (LowMemorySize >= BASE_1MB + PEI_MEM_SIZE);
-  PeiMemBase = (LowMemorySize - PEI_MEM_SIZE) & (~(BASE_64KB - 1));
-  DEBUG ((DEBUG_INFO, "UsableLowMemTop 0x%lx\n", LowMemorySize));
-  DEBUG ((DEBUG_INFO, "SystemLowMemTop 0x%x\n", PldMemInfo.SystemLowMemTop));
+  ASSERT (UsableLowMemTop >= BASE_1MB + PEI_MEM_SIZE);
+  PeiMemBase = (UsableLowMemTop - PEI_MEM_SIZE) & (~(BASE_64KB - 1));
+  DEBUG ((DEBUG_INFO, "UsableLowMemTop 0x%lx\n", UsableLowMemTop));
   DEBUG ((DEBUG_INFO, "PeiMemBase: 0x%lx.\n", PeiMemBase));
   DEBUG ((DEBUG_INFO, "PeiMemSize: 0x%lx.\n", PEI_MEM_SIZE));
   Status = PeiServicesInstallPeiMemory (PeiMemBase, PEI_MEM_SIZE);
   ASSERT_EFI_ERROR (Status);
 
-  //
-  // Set cache on the physical memory
-  //
-  MtrrSetMemoryAttribute (BASE_1MB, PldMemInfo.SystemLowMemTop - BASE_1MB, CacheWriteBack);
-  MtrrSetMemoryAttribute (0, 0xA0000, CacheWriteBack);

```

Comment:
Removal of this code could have been avoided by porting the change to disabel MTRR programming:

Reply:
> This feature was create for VP2410 initially, because changing MTRRs by EDK2 on this platform caused a hang. So if we ever make a new release for VP2410 using the rebased EDK2, it will likely not boot right now.

The 2 sentences are seemingly in contradiction,  conditional use of `MtrrLibNull` with coreboot implies that EDK2 should set MTRRs sometimes.  What feature was added for VP2410, use of `MtrrLibNull` or `DISABLE_MTRR_PROGRAMMING` define?

Reply:
> Are there cases where we would want EDK2 to set them?

I think there shouldn't be any cases with coreboot.

Reply:
`EFI_CPU_ARCH_PROTOCOL` used in https://github.com/Dasharo/edk2/blob/dasharo/MdeModulePkg/Core/Dxe/Misc/MemoryProtection.c seems to set MTRRs too.  But if `MtrrLibNull`  was used without trouble before, sounds like it's OK to always use it in `DasharoPayloadPkg`.

I don't see much use of returning code which does nothing and can't even work properly though.  Its counterpart in `UefiPayloadPkg` doesn't exist anymore, so we might as well simplify this code to suit our needs and look less confusing.

Reply:
For reference, this is how it is done in MrChromebox fork: https://github.com/MrChromebox/edk2/commit/8ea0ac749ff4cde85305544f79787b82f764b97e